File: code/library/src/main/java/org/fbme/lib/iec61499/parser/Iec61499ConverterConfiguration.java
package org.fbme.lib.iec61499.parser;

import org.fbme.lib.iec61499.IEC61499Factory;
import org.fbme.lib.st.STFactory;
import org.jetbrains.annotations.NotNull;

public interface Iec61499ConverterConfiguration {

    @NotNull IEC61499Factory getEntryFactory();

    @NotNull STFactory getStEntryFactory();

    @NotNull BasicFBTypeConverter createBasicFbTypeConverter(@NotNull ConverterArguments arguments);

    @NotNull CompositeFBTypeConverter createCompositeFbTypeConverter(@NotNull ConverterArguments arguments);
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/parser/Iec61499ConverterConfiguration.kt
package org.fbme.lib.iec61499.parser

import org.fbme.lib.iec61499.IEC61499Factory
import org.fbme.lib.st.STFactory

interface Iec61499ConverterConfiguration {
    val entryFactory: IEC61499Factory
    val stEntryFactory: STFactory
    fun createBasicFbTypeConverter(arguments: ConverterArguments): BasicFBTypeConverter
    fun createCompositeFbTypeConverter(arguments: ConverterArguments): CompositeFBTypeConverter
}